deploy.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+set -e # Exit with nonzero exit code if anything fails
+
+# From https://gist.github.com/domenic/ec8b0fc8ab45f39403dd
+
+echo "[TRACE] TRAVIS_BRANCH: $TRAVIS_BRANCH"
+echo "[TRACE] TRAVIS_PULL_REQUEST_BRANCH: $TRAVIS_PULL_REQUEST_BRANCH"
+echo "[TRACE] TRAVIS_PULL_REQUEST: $TRAVIS_PULL_REQUEST"
+
+SOURCE_BRANCH="$TRAVIS_PULL_REQUEST_BRANCH"
+TARGET_BRANCH="$TRAVIS_PULL_REQUEST-$TRAVIS_PULL_REQUEST_BRANCH"
+
+if [ "$TRAVIS_PULL_REQUEST" = "false" -a "$TRAVIS_BRANCH" != "master" ]; then
+   echo "[ABORT] We're in a push build (not in master nor in a pull request), so exiting. "
+   exit 0
+fi
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" -a "$TRAVIS_PULL_REQUEST_BRANCH" = "master" ]; then
+   echo "[ERROR] We're in a pull request but we're in the master branch?!?!"
+   echo " This shouldn't happen..."
+   exit 1
+fi
+
+if [ "$TRAVIS_PULL_REQUEST" = "false"  -a "$SOURCE_BRANCH" = "master" ]; then
+  SOURCE_BRANCH="master"
+  TARGET_BRANCH="gh-pages"
+fi
+
+echo "[TRACE] SOURCE_BRANCH: $SOURCE_BRANCH"
+echo "[TRACE] TARGET_BRANCH: $TARGET_BRANCH"
+
+# Save some useful information
+REPO=`git config remote.origin.url`
+SSH_REPO=${REPO/https:\/\/github.com\//git@github.com:}
+SHA=`git rev-parse --verify HEAD`
+
+# Clone the existing gh-pages for this repo into out/
+# Create a new empty branch if the target branch doesn't exist yet (should only happen on first deply)
+git clone $REPO out
+cd out
+git checkout $TARGET_BRANCH || git checkout --orphan $TARGET_BRANCH
+cd ..
+
+# Clean out existing contents
+rm -rf out/**/* || exit 0
+
+# Copy content from build into  existing contents
+
+echo -e "\n[TRACE] Building the specification"
+cd spec
+ant build
+
+# Make sure we're in the right directory
+
+echo -e "\n[TRACE] Copying the specification and cleaning out"
+
+cp -R build/* ../out/
+
+cd ../out
+
+rm -f .travis.yml
+rm -f w3c.json
+rm -f CODE_OF_CONDUCT.md
+rm -f CONTRIBUTING.md
+rm -f LICENSE.md
```

Comment:
Why do we remove `CODE_OF_CONDUCT.md`, `CONTRIBUTING.md` and `LICENSE.md`?

Reply:
Those remains in the master branch and the PR branches. I don't we need them in the automatic PR build branches imho.

Reply:
OK, not a big deal for me.

Reply:
@plehegar I do not understand why these files need to be removed since the build directory does not contain them?

Reply:
@plehegar What do you mean by "the original repo branch"?

Reply:
@plehegar More specifically, as a result of `cp -R build/* ../out/` , `../out` should not contain any spurious files since `ant build` only copies relevant files in `build`.

Reply:
the build doesn't but the new branch does, thus the clean up before committing.